todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java

```diff
@@ -30,78 +28,117 @@
  */
-public class AddEditTaskFragment extends Fragment implements AddEditTaskContract.View {
+public class AddEditTaskFragment extends Fragment {
 
     public static final String ARGUMENT_EDIT_TASK_ID = "EDIT_TASK_ID";
 
-    private AddEditTaskContract.Presenter mPresenter;
-
+    @Nullable
     private TextView mTitle;
 
+    @Nullable
     private TextView mDescription;
 
-    private String mEditedTaskId;
+    @Nullable
+    private CompositeSubscription mSubscription;
+
+    @Nullable
+    private AddEditTaskViewModel mViewModel;
 
     public static AddEditTaskFragment newInstance() {
         return new AddEditTaskFragment();
     }
 
-    public AddEditTaskFragment() {
-        // Required empty public constructor
-    }
-
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.subscribe();
+        bind();
     }
 
     @Override
     public void onPause() {
+        unbind();
         super.onPause();
-        mPresenter.unsubscribe();
     }
 
-    @Override
-    public void setPresenter(@NonNull AddEditTaskContract.Presenter presenter) {
-        mPresenter = checkNotNull(presenter);
+    private void bind() {
+        mSubscription = new CompositeSubscription();
+
+        mSubscription.add(getViewModel().getTask()
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Action1<Task>() {
+                    @Override
+                    public void call(Task task) {
+                        setTask(task);
+                    }
+                }, new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        showEmptyTaskError();
+                    }
+                }));
+    }
+
+    private void unbind() {
+        getSubscription().unsubscribe();
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
-        setTaskIdIfAny();
+        mViewModel = Injection.provideAddEditTaskViewModel(getContext(), getTaskId());
 
         FloatingActionButton fab =
                 (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task_done);
         fab.setImageResource(R.drawable.ic_done);
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (isNewTask()) {
-                    mPresenter.createTask(
-                            mTitle.getText().toString(),
-                            mDescription.getText().toString());
-                } else {
-                    mPresenter.updateTask(
-                            mTitle.getText().toString(),
-                            mDescription.getText().toString());
-                }
-
+                saveTask();
```

Comment:
Similar question as in other PR, if you want to have all reactive thing in bind, you can make an observable from click events. But is up to you